tests/TestCase/Model/Table/TagsTableTest.php

```diff
@@ -159,8 +159,25 @@ public function testAdminDoesRemoveTag() {
 
     public function testAddTag_correctDateUsingArabicLocale() {
         I18n::setLocale('ar');
-        $added = $this->Tag->addTag('arabic', 4, 1);
+        $added = $this->Tag->addTag('arabic', 4);
         $returned = $this->Tag->get($added->id);
-        $this->assertEquals($added->added_time, $returned->created);
+        $this->assertEquals($added->created, $returned->created);
+    }
+
+    public function testAddTagWithoutSentenceId_NoDuplicateAdded() {
+        $added = $this->Tag->addTag('regional', 4);
+        $this->assertNotEquals($added->user_id, 4);
+    }
+
+    public function testAddTag_addEmptyTag() {
+        $added = $this->Tag->addTag('', 1);
+        $this->assertFalse($added);
+    }
+
+    public function testAddTag_noTrailingSpaceAfterCuttingTo50Bytes() {
+        $tagName = '1234567890123456789012345678901234567890123456789 content after 9 gets cut';
+        $added = $this->Tag->addTag($tagName, 4);
+        $storedId = $this->Tag->getIdFromName(substr($tagName, 0, 49));
```

Comment:
Could you rewrite this test without using substr? In general it’s better to keep the tests as simple as we can because complexity is a source of bugs and because it’s easier to understand (I didn’t get the 49 at first).